itinerant-iterators/tests/test_obfuscate.py

```diff
@@ -0,0 +1,114 @@
+# flake8: noqa
+import tempfile
+from pathlib import Path
+
+import pytest
+from PIL import Image
+
+from app.obfuscate import ColourBox, validate_bounding_box
+
+
+def test_obfuscate_colour_box_colour_string():
+    # arrange
+    # create a new blank image with all white pixels
+    img = Image.new("RGB", (100, 100), (255, 255, 255))
+    # obfuscate by setting pixels to black
+    b = ColourBox("black")
+
+    # act
+    b.hide((25, 25, 75, 75), img)
+
+    # assert
+    all_pixels = list(img.getdata())
+    white = 0
+    black = 0
+    for pixel in all_pixels:
+        if pixel == (0, 0, 0):
+            black += 1
+        elif pixel == (255, 255, 255):
+            white += 1
+        else:
+            raise Exception(
+                f"obfuscated image contained pixel of unexpected colour: {pixel}"
+            )
+
+    assert img.size == (100, 100)
+    assert black == 2500
+    assert white == 7500
+
+
+def test_obfuscate_colour_box_colour_tuple():
+    # arrange
+    # create a new blank image with all white pixels
+    img = Image.new("RGB", (100, 100), (255, 255, 255))
+    # obfuscate by setting pixels to black
+    b = ColourBox((0, 0, 0))
+
+    # act
+    b.hide((25, 25, 75, 75), img)
+
+    # assert
+    all_pixels = list(img.getdata())
+    white = 0
+    black = 0
+    for pixel in all_pixels:
+        if pixel == (0, 0, 0):
+            black += 1
+        elif pixel == (255, 255, 255):
+            white += 1
+        else:
+            raise Exception(
+                f"obfuscated image contained pixel of unexpected colour: {pixel}"
+            )
+
+    assert img.size == (100, 100)
+    assert black == 2500
+    assert white == 7500
```

Comment:
This seem like a great candidate for [parametrized tests](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html). 

```python
@pytest.mark.parametrize(
    "obfuscator", [ColourBox("black"), ColourBox((0, 0, 0))], ids=["by_name", "by_tuple"]
)
def test_obfuscate_colour_box(obfuscator):
    # arrange
    # create a new blank image with all white pixels
    img = Image.new("RGB", (100, 100), (255, 255, 255))

    # act
    obfuscator.hide((25, 25, 75, 75), img)

    # assert
    all_pixels = list(img.getdata())
    white = 0
    black = 0
    for pixel in all_pixels:
        if pixel == (0, 0, 0):
            black += 1
        elif pixel == (255, 255, 255):
            white += 1
        else:
            raise Exception(
                f"obfuscated image contained pixel of unexpected colour: {pixel}"
            )

    assert img.size == (100, 100)
    assert black == 2500
    assert white == 7500
```
Also, perhaps the output could be checked against a known-good image file? That's real nickpicky though :p